pkg/porter/porter.go

```diff
@@ -26,6 +27,7 @@ type Porter struct {
 	Builder         BuildProvider
 	Manifest        *manifest.Manifest
 	Mixins          mixin.MixinProvider
+	Plugins         plugins.PluginProvider
```

Comment:
When we add interfaces to the Porter app, at the same time it really helps to add a test struct so that we can write unit tests and not end up hitting the file system and expect real plugins to be there.

So let's follow the same pattern that we used for mixins:

1. Add a TestPluginProvider using [TestMixinProvider](https://github.com/deislabs/porter/blob/d904e39007954fbec5e4d870e13ad16b04c355e2/pkg/mixin/helpers.go#L11) as an example. Though we don't need the run assertions stuff that it does, just whatever stubbing is necessary so that we could call `PrintPlugins` for example and have it print out a hard coded plugin like 'myplugin'.
1. Update the [TestPorter](https://github.com/deislabs/porter/blob/d904e39007954fbec5e4d870e13ad16b04c355e2/pkg/porter/helpers.go#L39) struct to use the new test stub by default.
2. Add a test for PrintPlugins that uses the new test plugin provider, using [TestPrintMixins](https://github.com/deislabs/porter/blob/d904e39007954fbec5e4d870e13ad16b04c355e2/pkg/porter/mixins_test.go#L12) as an example.